src/transformers/models/efficientformer/configuration_efficientformer.py

```diff
@@ -0,0 +1,165 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" EfficientFormer model configuration"""
+
+from typing import List
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+EFFICIENTFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "huggingface/efficientformer-l1": "https://huggingface.co/huggingface/efficientformer-l1/resolve/main/config.json",
+}
+
+
+class EfficientFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`EfficientFormerModel`]. It is used to instantiate
+    an EfficientFormer model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the EfficientFormer
+    [snap-research/efficientformer-l1](https://huggingface.co/snap-research/efficientformer-l1) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        layers (`list(int)`, *optional*, defaults to [3, 2, 6, 4])
+            Depth of each stage.
+        embed_dims (`list(int)`, *optional*, defaults to [48, 96, 224, 448])
+            Dimensionality of each stage.
+        downsamples (`list(bool)`, *optional*, defaults to [True, True, True, True])
```

Comment:
\`list(xxx)\`->\`List[xxx]\` to be consistent with other type annotation, and all the defaults that are lists needs to be in code formatting (so \`[3, 2, 6, 4]\` for instance)